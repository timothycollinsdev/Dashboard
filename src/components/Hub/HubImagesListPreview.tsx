import React from "react";
import ImageCard from "./ImageCard";
import styled from "@emotion/styled";

const SubTitle = styled.span`
  font-weight: 500;
  font-size: 1.5rem;
`;

const HubImagesListPreview = () => {
  // Todo: update images after fetching it from hub API
  const images = [
    {
      name: "Dummy MWU Encoder",
      id: "uudi1",
      author: "dev-team@jina.ai",
      description: "a minimum working unit of a containerized encoder",
      tags: ["Search", "encode"],
    },
    {
      name: "Dummy MWU Encoder",
      id: "uudi2",
      author: "dev-team@jina.ai",
      description: "a minimum working unit of a containerized encoder",
      tags: ["Search", "encode"],
    },
  ];

  return (
    <>
      <SubTitle>Latest</SubTitle>
      {images.map((image) => (
        <ImageCard image={image} key={image.id} />
      ))}
    </>
  );
};

export default HubImagesListPreview;
