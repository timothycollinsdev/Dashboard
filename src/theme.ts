const PERSIAN_GREEN = "#009999";
const GOLDEN_TANOI = "#FFCC66";
const BITTER_SWEET = "#ff6666";
const SCOOTER_GREEN = "#32C8CD";
const HEADER_TEXT_COLOR = "#000000";

const theme = {
  palette: {
    background: { default: "#FFFFFF" },
    headerTextColor: HEADER_TEXT_COLOR,
    primary: PERSIAN_GREEN,
    secondary: GOLDEN_TANOI,
    warning: GOLDEN_TANOI,
    error: BITTER_SWEET,
    success: SCOOTER_GREEN,
    info: PERSIAN_GREEN,
    critical: "#ff4540",
    debug: "#7E7278",
  },
};

export { theme };
