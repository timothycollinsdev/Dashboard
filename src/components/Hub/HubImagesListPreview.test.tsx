import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "@emotion/react";
import { theme } from "../../theme";
import HubImagesListPreview from "./HubImagesListPreview";
import { HashRouter as Router, Route } from "react-router-dom";

test("HubImagesListPreview", () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Router>
        <HubImagesListPreview />
      </Router>
    </ThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
