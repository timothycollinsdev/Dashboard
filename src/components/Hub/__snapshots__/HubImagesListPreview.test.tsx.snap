// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubImagesListPreview 1`] = `
Array [
  <span
    className="css-1gt7phv"
  >
    Latest
  </span>,
  <div
    className="clickable mb-4 h-100 image-card card"
  >
    <div
      className="pb-0 mb-0 pt-3 card-body"
    >
      <div
        className="row"
      >
        <div
          className="px-0 col-12"
        >
          <div
            className="app-title mb-2"
          >
            Dummy MWU Encoder
          </div>
          <div
            className="app-category mb-2"
          >
            Search
            encode
          </div>
          <div
            className="app-subtitle mb-2"
          >
            dev-team@jina.ai
          </div>
        </div>
        <div
          className="px-0 pb-0 col-sm-12"
        >
          <div
            className="description-container"
          >
            <div
              className="description-overlay"
            />
            <div
              className="app-description"
            >
              a minimum working unit of a containerized encoder
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="clickable mb-4 h-100 image-card card"
  >
    <div
      className="pb-0 mb-0 pt-3 card-body"
    >
      <div
        className="row"
      >
        <div
          className="px-0 col-12"
        >
          <div
            className="app-title mb-2"
          >
            Dummy MWU Encoder
          </div>
          <div
            className="app-category mb-2"
          >
            Search
            encode
          </div>
          <div
            className="app-subtitle mb-2"
          >
            dev-team@jina.ai
          </div>
        </div>
        <div
          className="px-0 pb-0 col-sm-12"
        >
          <div
            className="description-container"
          >
            <div
              className="description-overlay"
            />
            <div
              className="app-description"
            >
              a minimum working unit of a containerized encoder
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
